internal/reconciler/logpipeline/status.go

```diff
@@ -61,24 +62,23 @@ func (r *Reconciler) updateStatusUnsupportedMode(ctx context.Context, pipeline *
 }
 
 func (r *Reconciler) setAgentHealthyCondition(ctx context.Context, pipeline *telemetryv1alpha1.LogPipeline) {
-	healthy, err := r.agentProber.IsReady(ctx, r.config.DaemonSet)
-	if err != nil {
-		logf.FromContext(ctx).V(1).Error(err, "Failed to probe fluent bit daemonset - set condition as not healthy")
-		healthy = false
-	}
+	status := metav1.ConditionTrue
+	reason := conditions.ReasonAgentReady
+	msg := conditions.MessageForLogPipeline(reason)
 
-	status := metav1.ConditionFalse
-	reason := conditions.ReasonAgentNotReady
-	if healthy {
-		status = metav1.ConditionTrue
-		reason = conditions.ReasonAgentReady
+	err := r.agentProber.IsReady(ctx, r.config.DaemonSet)
+	if err != nil && !workloadstatus.IsRolloutInProgressError(err) {
+		logf.FromContext(ctx).V(1).Error(err, "Failed to probe fluent bit daemonset - set condition as not healthy")
+		status = metav1.ConditionFalse
+		reason = conditions.ReasonAgentNotReady
+		msg = err.Error()
 	}
 
 	condition := metav1.Condition{
 		Type:               conditions.TypeAgentHealthy,
 		Status:             status,
 		Reason:             reason,
-		Message:            conditions.MessageForLogPipeline(reason),
+		Message:            msg,
 		ObservedGeneration: pipeline.Generation,
 	}
 
```

Comment:
This code looks quite similar to the one here: internal/reconciler/logparser/status.go:40
we should make this a reusable function